tests/Unit/ActionTesting.hpp

```diff
@@ -324,6 +327,16 @@ class MockArrayElementProxy {
     tuples::get<InboxTag>(inbox_)[id].emplace(data);
   }
 
+  template <typename Action, typename... Args>
```

Comment:
I would have been able to anticipate the presence and motivation for these functions if we had in mind the same definition for "proxy" - I only had in mind a general, more vague definition of "proxy" while you were adopting a much more precise agreed-upon definition. This is also the case with "mock" - considering there are many classes in SpECTRE that are given names with various levels of preciseness, 

We have agreed that it is good to preface classes like this with a comment stating why the class is named such.